tests/basetest.rb

```diff
@@ -137,7 +137,23 @@ def teardown
     @device = nil
   end
 
+  # Execute the specified config commands and warn if the
+  # output matches the default "warning" regex.
   def config(*args)
+    config_and_warn_on_match(/^invalid|^%/i, *args)
```

Comment:
I see you went with the anchored regexp.  @glennmatthews Any comments on this change?